Paltineanu Rares-Mihai/curs9 - 3.8/main.c

```diff
@@ -0,0 +1,114 @@
+/** Read  from  the standard input the  author,  the  title  and  the  publication  year  for  a  number  of n books.
+Display the following:
+a)the names of the authors in alphabetic order;
+b)the names of the authors and the titles of their books in order of the publication year.
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+struct probl {
+
+    char author[200];
+    char title[200];
+    int year;
+
+}x[200], aux;
+
+void read(int n) {
+
+    for(int i = 0; i < n; i++) {
+
+        printf("Read the author: "); gets(x[i].author);
+        printf("Read the title: "); gets(x[i].title);
+        printf("Read the year: "); scanf("%d", &x[i].year);
+        getchar();
+        printf("\n");
+
+    }
+}
+
+void alphabeticOrder(int n) {
```

Comment:
Here's a challenge for you.

You see these two functions, `alphabeticOrder` and `publicationYear` do the same thing really, they sort the array but they sort it based on different things, and the only difference code-wise is that compare function, right?
`alphabeticOrder` compares based on author, while `publicationYear` compares based on year. 

Based on https://codeforwin.org/2017/12/pass-function-pointer-as-parameter-another-function-c.html
Could you find a way to have just one function and it to take as a parameter another function which will be make this comparison?

Food for thought!


Oh, and by the way - crappy names for these sorting functions, they should be something like `sortByAuthor` and `sortByYear` - they are functions that *do* something!

Reply:
I will try it, hope that i will not forgot